tests/unit/calendar/options.js

```diff
@@ -232,7 +235,81 @@ test( "min / max", function() {
 		.calendar( "value", "1/4/09" );
-	testHelper.equalsDate( element.calendar( "valueAsDate" ), new Date( 2008, 6 - 1, 4 ), "Min/max - value > max" );});
+	testHelper.equalsDate( element.calendar( "valueAsDate" ), new Date( 2008, 6 - 1, 4 ), "Min/max - value > max" );
+
+	element
+		.calendar( "option", { min: minDate, max: maxDate } )
+		.calendar( "value", "3/4/08" );
+	ok( !prevButton.hasClass( "ui-state-disabled" ), "Prev button enabled" );
+	prevButton.simulate( "click" );
+	ok( prevButton.hasClass( "ui-state-disabled" ), "Prev button disabled" );
+
+	element.calendar( "value", "11/4/08" );
+	ok( !nextButton.hasClass( "ui-state-disabled" ), "Next button enabled" );
+	nextButton.simulate( "click" );
+	ok( nextButton.hasClass( "ui-state-disabled" ), "Next button disabled" );
+
+	element
+		.calendar( "option", { max: null } )
+		.calendar( "value", "1/4/09" )
+		.calendar( "option", { min: minDate, max: maxDate } );
+	ok( nextButton.hasClass( "ui-state-disabled" ), "Other year above max: Next button disabled" );
+	prevButton.simulate( "click" );
+	ok( nextButton.hasClass( "ui-state-disabled" ), "Other year above max: Next button disabled after click" );
+	prevButton.simulate( "click" );
+	ok( !nextButton.hasClass( "ui-state-disabled" ), "Other year above max: Next button enabled after click" );
+
+	element
+		.calendar( "option", { min: null } )
+		.calendar( "value", "1/4/08" )
+		.calendar( "option", { min: minDate, max: maxDate } );
+	ok( prevButton.hasClass( "ui-state-disabled" ), "Other year below min: Prev button disabled" );
+	nextButton.simulate( "click" );
+	ok( prevButton.hasClass( "ui-state-disabled" ), "Other year below min: Prev button disabled after click" );
+	nextButton.simulate( "click" );
+	ok( !prevButton.hasClass( "ui-state-disabled" ), "Other year below min: Prev button enabled after click" );
+});
+
+test( "numberOfMonths", function() {
+	expect( 6 );
+	var date = new Date( 2015, 8 - 1, 1 ),
+		input = $( "#calendar" ).calendar({
+			numberOfMonths: 3,
+			value: date
+		}),
+		container = input.calendar( "widget" );
+
+	equal( container.find( ".ui-calendar-group" ).length, 3, "3 calendar grids" );
+	equal(
+		container.find( "tbody:first td[id]:first" ).attr( "id" ),
+		"calendar-2015-7-1",
+		"Correct id set for first day of first grid"
+	);
+	equal(
+		container.find( "tbody:last td[id]:last" ).attr( "id" ),
+		"calendar-2015-9-31",
+		"Correct id set for last day of third grid"
+	);
+
+	// Test for jumping in weekday rendering after click on last day of last grid
+	container.find( "tbody:last td[id]:last button" ).trigger( "mousedown" );
+	equal( container.find( "thead:last th:last" ).text(), "Sa",
+		"After mousedown last month: Last day is Saturday"
+	);
+
+	// Test if using cursor to go to the next / prev month advances three month
+	container.find( "tbody:first td[id]:first button" ).trigger( "mousedown" );
+	$( document.activeElement ).simulate( "keydown", { keyCode: $.ui.keyCode.LEFT } );
```

Comment:
We need to focus the first / last day of a month, then using arrow keys to jump to prev / next month. Could probably use plain focus on that first / last day.